content/courses/advanced-ada/chapters/subprograms_modularity.rst

```diff
@@ -85,7 +96,53 @@ In this example, we declare :ada:`Calculate` as a private procedure of the
 :ada:`Data_Processing` package. Therefore, it's visible in that package (but
 not in the :ada:`Test_Data_Processing` procedure).
 
-.. admonition:: In the Ada Reference Manual
 
-    - :arm:`10.1.1 Compilation Units - Library Units <10-1-1>`
-    - :arm:`10.1.2 Context Clauses - With Clauses <10-1-2>`
+Private subprograms and private packages
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+We can also use private subprograms to test private packages. As we know, in
+most cases, we cannot access private packages in external clients |mdash| such
+as external subprograms. However, by declaring a subprogram private, we're
```

Comment:
Maybe you should emphasize that these must be library-level parameterless subprograms, because those can be main programs.

Reply:
Yes, although readers should already be aware of this requirement at this point, it's good to mention it.